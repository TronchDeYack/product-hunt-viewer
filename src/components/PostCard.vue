<template>
  <div class="post-card">
    <img class="card-img" :src="img" />
    <div class="card-body">
      <h3 class="title">{{ title }}</h3>
      <p class="upvotes">{{ votes }} upvotes</p>
      <p class="description">{{ description }}</p>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.post-card {
  width: 20rem;
  background: white;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
  transition: box-shadow 0.3s ease-in-out;
  margin: 2rem;

  &:hover {
    cursor: pointer;
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.5);
  }

  .card-img {
    height: 180px;
    width: 100%;
    display: block;
  }

  .card-body {
    flex: 1 1 auto;
    padding: 0 1.25rem 1.25rem 1.25rem;

    .title {
      margin-bottom: .75rem;
    }
    .upvotes {
      font-size: 12px;
    }
  }
}
</style>

<script>
export default {
  name: 'post-card',
  props: {
    img: String,
    title: String,
    votes: Number,
    description: String,
  },
};
</script>
