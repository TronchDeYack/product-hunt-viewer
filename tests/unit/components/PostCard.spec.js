import { mount } from '@vue/test-utils';

import PostCard from '@/components/PostCard.vue';

describe('PostCard', () => {
  it('Should render.', () => {
    const wrapper = mount(PostCard, {
      propsData: {
        img: 'https://hello.com/img/random.png',
        title: 'Kite compositor',
        votes: 731,
        description: 'This is a post about interesting things... I guess !',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
