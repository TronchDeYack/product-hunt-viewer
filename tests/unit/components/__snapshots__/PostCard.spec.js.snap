// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard Should render. 1`] = `
<div class="post-card"><img src="https://hello.com/img/random.png" class="card-img">
  <div class="card-body">
    <h3 class="title">Kite compositor</h3>
    <p class="upvotes">731 upvotes</p>
    <p class="description">This is a post about interesting things... I guess !</p>
  </div>
</div>
`;
